packages/contracts/contracts/L1/ValidatorManager.sol

```diff
@@ -129,6 +129,14 @@ contract ValidatorManager is ISemver, IValidatorManager {
      */
     mapping(uint256 => uint128) internal _pendingChallengeReward;
 
+    /**
+     * @notice Output index only allowed to submit to TrustedValidator.
+                Challenging this output is also restricted.
+     */
+    uint256 public restrictedOutputIndex;
+
+    uint256[49] private __gap;
```

Comment:
Why did you insert gaps?

Reply:
The restrictedOutputIndex is a temporary field for MPT transitions, so I envision removing it in a future upgrade.

Reply:
Then you can store that value with randomness key.
```solidity
    bytes32 internal constant RESTRICTED_OUTPUT_INDEX_KEY = <keccak256('restrictedOutputIndex`)>;
    ...
    function setRestrictedOutputIndex(uint256 _outputIndex) public onlyTrustedValidator {
        assembly {
            sstore(RESTRICTED_OUTPUT_INDEX_KEY, _outputIndex)
        }
    }
    
    function getRestrictedOutputIndex() public view returns (uint256) {
        uint256 index;
        assembly {
            index := sload(RESTRICTED_OUTPUT_INDEX_KEY)
        }
        return index;
    }
```